pkg/dns/dns.go

```diff
@@ -19,42 +19,66 @@ import (
 	"fmt"
 	"os"
 	"regexp"
+	"strings"
 )
 
 const (
 	// Default location for the resolv.conf file.
 	DefaultResolveConfPath = "/etc/resolv.conf"
 
 	// Default cluster domain value for k8s clusters.
-	DefaultLocalDNS = "svc.cluster.local"
+	DefaultClusterDomain = "cluster.local"
 )
 
-// GetLocalDNSName parses the path to resolv.conf to find the local DNS name.
-func GetLocalDNSName(resolvConfPath string) (string, error) {
-	var localDNSName string
+// GetClusterDomain parses the path to resolv.conf to find the cluster domain.
+func GetClusterDomain(resolvConfPath string) (string, error) {
+	var clusterDomain string
 	file, err := os.Open(resolvConfPath)
 	if err != nil {
 		return "", err
 	}
 	defer file.Close()
 
-	reg := regexp.MustCompile(`^search.*?\s(svc\.[^\s]*)`)
+	reg := regexp.MustCompile(`^search.*?\ssvc\.([^\s]*)`)
 
 	scanner := bufio.NewScanner(file)
 	for scanner.Scan() {
 		match := reg.FindStringSubmatch(scanner.Text())
 		if len(match) > 0 {
-			localDNSName = match[1]
+			clusterDomain = match[1]
 		}
 	}
 
 	if err := scanner.Err(); err != nil {
 		return "", err
 	}
 
-	if localDNSName == "" {
-		return "", fmt.Errorf("failed to find local DNS name in resolv.conf")
+	if clusterDomain == "" {
+		return "", fmt.Errorf("failed to find cluster domain in resolv.conf")
 	}
 
-	return localDNSName, nil
+	return clusterDomain, nil
+}
+
+// GetServiceDNSNames parses the fully-qualified domain service name nd
+// returns a list of its qualified service names.
+func GetServiceDNSNames(fqdnServiceName string, clusterDomain string) ([]string, error) {
+	all := strings.Split(fqdnServiceName, ".")
+
+	// The FQDN service name will be: <svc_name>.<ns>.svc.<cluster-domain>
+	// There is no guarantee the cluster-domain will be dotted (kubelet does not
+	// validate the cluster domain).
+	if len(all) < 4 {
+		return nil, fmt.Errorf("failed to split FQDN service name %q", fqdnServiceName)
+	}
+
+	// We return:
+	// - <svc_name>.<ns>
```

Comment:
Do we need to handle the local-namespace access case as well? e.g., just `<svc_name>`?

Reply:
We don't need it but I'll add it for completeness.